From 04d0174caaa7fa3204623461a350c25990402d15 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 21 Aug 2022 00:42:25 +0200
Subject: [PATCH 6/6] arm64: dts: rk3399-pinephone-pro: Fix initial codec clock
 frequency

By default, I2S output clock is 800 Mhz, which is too much for the
codec. Change the initial frequency to 256 * 44.1 kHz.

This fixes codec not working right after boot and before first
audio file is played.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 719585c4bb5e..54947c39cbb3 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -944,6 +944,8 @@ rt5640: rt5640@1c {
 		reg = <0x1c>;
 		clocks = <&cru SCLK_I2S_8CH_OUT>;
 		clock-names = "mclk";
+		assigned-clocks = <&cru SCLK_I2S0_8CH>;
+		assigned-clock-rates = <11289599>;
 		realtek,dmic1-data-pin = <1>;
 		realtek,in3-differential;
 		realtek,lout-differential;
-- 
2.35.3

